firehose.c

```diff
@@ -401,6 +403,9 @@ static int firehose_program(struct qdl_device *qdl, struct program *program, int
 	lseek(fd, (off_t) program->file_offset * program->sector_size, SEEK_SET);
 	left = num_sectors;
 	while (left > 0) {
+		fprintf(stderr, "[PROGRAM] %d sectors remaining out of %d (%.2f%%)\r",
```

Comment:
I was thinking that we didn't want to update this too often, but with the typical 1MB max_payload_size, and if my math is correct, this will generate about 10kb of logs per second on my machine. So, I don't think we need to make this more fancy at this time.

I'm thinking that we should clean up the logs a bit in general. Until then, I like this patch!

But perhaps move it last in the loop? That way it will end at 100%.

Reply:
For larger images QDL appear to be stuck without any logs. So updated the patch to limit to 1 log per second. Do check if this is fine.